scripts/joinmarket-qt.py

```diff
@@ -698,9 +735,36 @@ def qt_directsend_callback(rtxd, rtxid, confs):
                 self.cleanUp()
             return
 
+        if bip78url:
+            # this is needed for manual regtest GUI testing
+            tls_whitelist_for_testing = [b"127.0.0.1"]
+            manager = parse_payjoin_setup(self.bip21_uri,
+                mainWindow.wallet_service, mixdepth, "joinmarket-qt")
+            # disable form fields until payment is done
+            self.addressInput.setEnabled(False)
+            self.pjEndpointInput.setEnabled(False)
+            self.mixdepthInput.setEnabled(False)
+            self.amountInput.setEnabled(False)
+            self.startButton.setEnabled(False)
+            d = task.deferLater(reactor, 0.0, send_payjoin, manager,
+                    accept_callback=self.checkDirectSend,
+                    info_callback=self.infoDirectSend,
+                    tls_whitelist=tls_whitelist_for_testing)
```

Comment:
Only apply the whitelist when we are on regtest (check `isinstance(jm_single().bc_interface, RegtestBitcoinCoreInterface` iirc), to be safe.

Reply:
Wait, I just had another thought about this, this is none of Qt's business; we can put this check in `jmclient.payjoin.py` instead. Even more, we should be putting all blockchain logic code inside the walletservice, since we really want to decouple all the client stuff from anything low level like blockchain for easier changes down the line. So either put this check in payjoin.py via the sendpayment function, and/or have that function query the wallet service for whether we are on regtest or not.